### PR TITLE
Restructure how-to docs to categorise per platform

### PR DESCRIPTION
**Pull request by enxebre**
**What this PR does / why we need it**:
Restructure how-to docs to categorise per platform.
Follow up for https://github.com/openshift/hypershift/pull/1079

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [x] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: docs/content/how-to/none/create-none-cluster.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
-# Bare Metal workers using 'none' provider
+# Create a None cluster
 
-This document explains how to deploy HyperShift in OpenShift and then use the 'none' provider to deploy bare metal worker nodes for our `HostedClusters`.
+This document explains how to deploy a HyperShift cluster and then use the 'None' platform to deploy bare metal worker nodes for our `HostedClusters`.
```
Comment (e-minguez) [inline docs/content/how-to/none/create-none-cluster.md:3]:
```suggestion
This document explains how to deploy an HyperShift cluster and then use the 'None' platform to deploy bare metal worker nodes for our `HostedClusters`.
```

**Review by e-minguez — COMMENTED** [review]
two minor nits :)

**Comment by avishayt** [comment]
Looks good to me, thank you

File: docs/content/how-to/agent/create-agent-cluster.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
-# Agent platform on baremetal OCP
+# Create an Agent cluster
 
-This document explains how to deploy HyperShift in OpenShift and then use the 'agent' provider to deploy bare metal worker nodes for our `HostedClusters`.
+This document explains how create a HyperShift cluster using the 'Agent' platform to deploy bare metal worker nodes for our `HostedClusters`.
```
[the remark this answers is not in the document]
Comment (csrwng) [reply docs/content/how-to/agent/create-agent-cluster.md]: I believe `a Hypershift` is right (vs `an Hypershift`)

[the remark this answers is not in the document]
Comment (e-minguez) [reply docs/content/how-to/agent/create-agent-cluster.md]: [You are correct](https://www.a-or-an.com/a_an/hyper), my bad, sorry for the noise!

**Pull request merged**